droid-core-interfaces/src/main/java/uk/gov/nationalarchives/droid/core/interfaces/archive/ByteseekWindowWrapper.java

```diff
@@ -31,99 +31,99 @@
      * closeReaderIfClosed is true.
      *
      * @param reader The WindowReader backing this ReadOnlyFile.
      * @param closeReaderIfClosed If true, then the backing WindowReader will be closed when this is closed.
      */
-    public TrueZipReader(final WindowReader reader, final boolean closeReaderIfClosed) {
+    public ByteseekWindowWrapper(final WindowReader reader, final boolean closeReaderIfClosed) {
         super();
         this.reader = reader;
         this.closeReaderIfClosed = closeReaderIfClosed;
     }
 
+
     @Override
-    public long length() throws IOException {
-        return reader.length();
+    public int read(ByteBuffer dst) throws IOException {
+        ensureOpen();
+        if (currentPosition >= size()) {
+            return -1;
+        }
+        final int bytesCopied = ArchiveFileUtils.copyToBuffer(reader, currentPosition, dst);
+        currentPosition += bytesCopied;
+        return bytesCopied;
     }
 
     @Override
-    public long getFilePointer() throws IOException {
-        ensureOpen();
-        return filePointer;
+    public int write(ByteBuffer src) throws IOException {
+        throw new IOException(NOT_IMPLEMENTED);
     }
 
     @Override
-    public int read() throws IOException {
-        ensureOpen();
-        final int result = reader.readByte(filePointer);
-        if (result >= 0) {
-            filePointer++;
-        }
-        return result;
+    public long position() throws IOException {
+        return currentPosition;
     }
 
     @Override
-    public void seek(long position) throws IOException {
-        ensureOpen();
-        if (position < 0) {
-            throw new IOException("Cannot seek to a negative position: " + position);
-        }
-        if (position >= reader.length()) {
-            throw new IOException("Cannot seek past the end of data with length "
-                    + reader.length() + ".  Seek position was " + position);
-        }
-        filePointer = position;
+    public SeekableByteChannel position(long newPosition) throws IOException {
+        currentPosition = newPosition;
+        return this;
     }
 
     @Override
-    public int read(byte[] bytes, int offset, int length) throws IOException {
-        ensureOpen();
-        if ((offset | length) < 0) {
-            throw new IndexOutOfBoundsException("Offset or length cannot be negative: {" + offset + "," + length + "}");
-        }
-        if (filePointer >= length() || length == 0) {
-            return -1;
-        }
-        final int bytesCopied = ArchiveFileUtils.copyToBuffer(reader, filePointer, bytes, offset, length);
-        filePointer += bytesCopied;
-        return bytesCopied;
+    public long size() throws IOException {
+        return reader.length();
+    }
+
+    @Override
+    public SeekableByteChannel truncate(long size) throws IOException {
+        throw new IOException(NOT_IMPLEMENTED);
+    }
+
+    @Override
+    public boolean isOpen() {
```

Comment:
Is this method used anywhere as I can't see it in the original `TrueZipReader`? Or are we including this for parity with `SevenZipReader`?

If we are keeping it should it have some tests?